specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json

```diff
@@ -4392,9 +4392,48 @@
             "type": "string"
           },
           "description": "The API versions."
+        },
+        "pattern": {
+          "$ref": "#/definitions/AliasPatternType",
+          "description": "The pattern for an alias path."
         }
       },
-      "description": "The type of the paths for alias. "
+      "description": "The type of the paths for alias."
+    },
+    "AliasPatternType": {
+      "properties": {
+        "phrase": {
+          "type": "string",
+          "description": "The alias pattern phrase."
+        },
+        "variable": {
+          "type": "string",
+          "description": "The alias pattern variable."
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "NotSpecified",
+            "Extract"
+          ],
+          "x-ms-enum": {
+            "name": "AliasPatternType",
+            "modelAsString": false,
```

Comment:
Usually setting it to `true` can give us more extensibility.

Reply:
Any downside to using **true**? I noticed most other enums in this swagger use **false**.

Reply:
You can check here for detail. http://azure.github.io/autorest/extensions/#x-ms-enum

Reply:
Changed both new enums to "modelAsString": "true".

Reply:
Decided against this and reverted the last change. The specific integer values matter, so I think "modelAsString": "false" is the right approach.

Reply:
Integer values? These enums are serialized in JSON as strings regardless of this setting. All this changes is whether generated SDK uses a string or enum for that particular property.

---
In reply to: [383561534](https://github.com/Azure/azure-rest-api-specs/pull/8456#discussion_r383561534) [](ancestors = 383561534)